debug_module.tex

```diff
@@ -110,39 +110,22 @@ \section{Selecting Harts} \label{selectingharts}
 
 \section{Run Control} \label{runcontrol}
 
-The Debug Module can halt selected harts and allow them to run again using
-the \Rdmcontrol register.
-After selecting harts, it sets \Fhaltreq, then waits for \Fallhalted to
-indicate the harts are halted before clearing \Fhaltreq to 0. Setting \Fhaltreq
-has no effect on harts which are already halted.
-
-To resume, the debugger sets \Fresumereq. This action sets each selected hart's
-{\tt resumeack} bit to 0. Once a hart has resumed, it
-sets its {\tt resumeack} bit to 1. Thus, the
-debugger should wait for \Fallresumeack to indicate all
-harts have resumed before clearing \Fresumereq to 0.
-If a debugger reads \Fanyresumeack and \Fanyhalted in the same operation, then
-some selected harts must have resumed and then halted again.
-
-\begin{commentary}
-  When waiting for a hart to resume, a debugger should
-  examine \Fallresumeack, not \Fallrunning
-  or \Fallhalted,
-  because the hart may immediately halt again due to
-  trigger or step conditions.
-\end{commentary}
-
-Setting \Fresumereq on hart which is running
-clears {\tt resumeack} but
-may cause a hart which halts in the future to immediately
-resume. Debuggers should not depend on this behavior and should not
-set \Fresumereq for running harts.
-
-To halt or resume multiple harts at the same time, the debugger first
-sets the hart's bits in the hart array mask register, then
-follows the same procedure but with \Fhasel set to 1.
-Depending on the desired operation, the debugger might consider the {\tt any*}
-versions of the status instead of {\tt all*}.
+For every hart, the Debug Module contains 3 bits of state: halt request, resume
+request, and hart reset. (The hart reset bit is optional.) These bits all reset
+to 0. A debugger can write them for the currently selected harts through
+\Fhaltreq, \Fresumereq, and \Fhartreset in \Rdmcontrol. In addition the DM
+receives halted, running, and resume ack signals from each hart.
+
+When a running hart sees a halt request, it responds by halting and asserting
+its halted signal. The halted signals of all selected harts are reflected in
+the \Fallhalted and \Fanyhalted bits. \Fresumereq is ignored by running harts.
```

Comment:
Shouldn't the \Fresumereq and \Fhaltreq "is ignored by" be swapped? This one seems like it would go with the halt and the other seems like it shoudl go with the resume.

Reply:
Yes. Thanks.